STEP_Project/preferred.js

```diff
@@ -4,44 +4,87 @@ if (chrome.storage !== undefined)
     storageObj = chrome.storage.sync;
 }
 
+var Tabs;
+if (chrome.tabs !== undefined)
+{
+    Tabs = chrome.tabs;
+}
+
 /**
  * Accesses chrome storage and displays the form of preferred entities with radio buttons
  */
 function createForm()
 {
-    storageObj.get(null, function (data) 
-    { 
-        let keys = Object.keys(data);
-        let numKeys = keys.length;
+    Tabs.query({active: true, lastFocusedWindow: true}, function(tabs) 
+    {
+        let url = tabs[0].url;
+        storageObj.get(null, function (data) 
+        {
+            let urlRoot = url.split('/ac/')[0];
+            urlRoot = urlRoot + '/ac/';
 
-        let selectForm = document.getElementById('radioButtons');
+            let urlQueries = url.split('ac')[1].split('?')[1];
 
-        for (let i = 0; i < numKeys; i++)
-        {
-            let elt = document.createElement('input');
-            elt.setAttribute('type', 'radio');
-            elt.setAttribute('value', keys[i]);
-            elt.setAttribute('name', 'preferred');
-
-            let label = document.createElement('label');
-            label.appendChild(elt);
-            let OUName = document.createTextNode(data[keys[i]]);
-            label.appendChild(OUName);
-
-            selectForm.appendChild(label);
-            let breakLine = document.createElement('br');
-            selectForm.appendChild(breakLine);
-        }
+            let keys = Object.keys(data);
+            let numKeys = keys.length;
+
+            let selectForm = document.getElementById('radioButtons');
 
-        selectForm.addEventListener('click',enableApplyButton);
+            let entityToDisplay = data['entity-to-display'];
+
+            for (let i = 0; i < numKeys; i++)
```

Comment:
(optional) You can also use Object.values here.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values#:~:text=Description-,Object.,values%20of%20the%20object%20manually.

Reply:
In this particular case, I don't think I should do that because one of the keys is 'entity-to-display', which cannot be the same as the key of any other entity. However, the value of 'entity-to-display', which is either 'OU', 'group' or 'user' could also be the value for some other key (the name of some entity). So I think looping through the keys instead of values prevents that problem.

Reply:
In that case you can do:

Object.keys(obj).forEach((key) => {
   const value = obj[key];
});

Reply:
Made a commit incorporating this